main.cpp

```diff
@@ -108,7 +108,410 @@ write 3 UDTs below that EACH have:
+}
+//--
+void Classroom::printClassDetails(){
+    std::cout << "Classroom: " << classroomName << ", Total Students: " << totalStudents << std::endl;
+    std::cout << "Class Average Grade: " << averageGrade << " (" << averageLetterGrade << ")" << std::endl;
+    for (const Student& student : students) 
+    {
+        student.printDetails();
+    }
+}
+//--
+char Classroom::convertGradeIntoLetterGrade(double grade)
+{
+    if (grade >= 90) return 'A';
+    else if (grade >= 80) return 'B';
+    else if (grade >= 70) return 'C';
+    else if (grade >= 60) return 'D';
+    else return 'F';
+}
+//--
+struct RecreationCenter 
+{
+    Classroom academicClassroom;
+    Gym fitnessGym;
+
+    RecreationCenter(const Classroom& classroom, const Gym& gym);
+    ~RecreationCenter();
+
+    double compareAverageAge() const; //returns difference between average age of students in academic classroom and average age of members in fitness gym
+    void listUniqueNames() const;
+
+};
+//--
+RecreationCenter::RecreationCenter(const Classroom& classroom, const Gym& gym)
+    : academicClassroom(classroom), fitnessGym(gym)
+{
+    std::cout << "RecreationCenter constructor called!" << std::endl;
+}
+//--
+RecreationCenter::~RecreationCenter()
+{
+    std::cout << "RecreationCenter destructor called!" << std::endl;
+}
+//--
+double RecreationCenter::compareAverageAge() const 
+{
+    double classroomSum = 0;
+    int studentCount = 0;
+    for (const auto& student : academicClassroom.students) 
+    {
+        classroomSum += student.age;
+        studentCount++;
+    }
+
+    double gymSum = 0;
+    int memberCount = 0;
+    for (const auto& member : fitnessGym.members) 
+    {
+        gymSum += member.age;
+        memberCount++;
+    }
+
+    double classroomAverage = (studentCount > 0) ? classroomSum / studentCount : 0;
+    double gymAverage = (memberCount > 0) ? gymSum / memberCount : 0;
+
+    return classroomAverage - gymAverage;
+}
+//--
+void RecreationCenter::listUniqueNames() const 
+{
+    std::unordered_set<std::string> uniqueNames;
+
+    for (const auto& student : academicClassroom.students) 
+    {
+        uniqueNames.insert(student.name);
+    }
+
+    for (const auto& member : fitnessGym.members) 
+    {
+        uniqueNames.insert(member.name);
+    }
+
+    std::cout << "Unique Names in Recreation Center:" << std::endl;
+    for (const auto& name : uniqueNames) 
+    {
+        std::cout << name << std::endl;
+    }
+}
+//--
+struct LoanManager 
+{
+    std::vector<Loan> loans;
+
+    LoanManager();
+    ~LoanManager();
+
+    void addLoan(const Loan& loan);
+    double calculateTotalMonthlyPaymentForNumMonth(unsigned int num) const;
+};
+//--
+LoanManager::LoanManager()
+{
+    std::cout << "LoanManager constructor called!" << std::endl;
+}
+//--
+LoanManager::~LoanManager()
+{
+    std::cout << "LoanManager destructor called!" << std::endl;
+}
+//--
+void LoanManager::addLoan(const Loan& loan) 
+{
```

Comment:
'const' and references have not been introduced yet in the course.
please refrain from using language features that haven't been introduced in the course yet. 
- [ ] revise